src/backend/engine/index.js

```diff
@@ -1,3 +1,23 @@
-define(["./TestableFile"], function(Testable) {
-    alert("calculation result is " + new Testable(function() { return 3; }).activity());
+define(["mod_parser", "mod_executor", "mod_data_structures"], function(parser, executor, dataStructures) {
+    
+    var process;
+
+    function runProgram(source) {
```

Comment:
I agree, but this is more than compilation, it also loads the program to memory and puts backend in ready to execution state. How about `loadProgram`?